lib/ansible/plugins/callback/selective.py

```diff
@@ -0,0 +1,257 @@
+}
+
+DONT_COLORIZE = os.getenv('ANSIBLE_SELECTIVE_DONT_COLORIZE', default=False)
+
+
+def dict_diff(prv, nxt):
+    """Return a dict of keys that differ with another config object."""
+    keys = set(prv.keys() + nxt.keys())
+    result = {}
+    for k in keys:
+        if prv.get(k) != nxt.get(k):
+            result[k] = (prv.get(k), nxt.get(k))
+    return result
+
+
+def colorize(msg, color):
+    """Given a string add necessary codes to format the string."""
+    if DONT_COLORIZE:
+        return msg
+    else:
+        return '{}{}{}'.format(COLORS[color], msg, COLORS['endc'])
+
+
+class CallbackModule(CallbackBase):
+    """selective.py callback plugin."""
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'stdout'
+    CALLBACK_NAME = 'selective'
+
+    def __init__(self, display=None):
+        """selective.py callback plugin."""
+        super(CallbackModule, self).__init__(display)
+        self.last_skipped = False
+        self.last_task_name = None
+        self.printed_last_task = False
+
+    def _print_task(self, task_name=None):
+        if task_name is None:
+            task_name = self.last_task_name
+
+        if not self.printed_last_task:
+            self.printed_last_task = True
+            line_length = 120
+            if self.last_skipped:
+                print()
+            msg = colorize("# {} {}".format(task_name,
+                                            '*' * (line_length - len(task_name))), 'bold')
+            print(msg)
+
+    def _indent_text(self, text, indent_level):
+        lines = text.splitlines()
+        result_lines = []
+        for l in lines:
+            result_lines.append("{}{}".format(' '*indent_level, l))
+        return '\n'.join(result_lines)
+
+    def _print_diff(self, diff, indent_level):
+        if isinstance(diff, dict):
+            try:
+                diff = '\n'.join(difflib.unified_diff(diff['before'].splitlines(),
+                                                      diff['after'].splitlines(),
+                                                      fromfile=diff.get('before_header',
+                                                                        'new_file'),
+                                                      tofile=diff['after_header']))
+            except AttributeError:
+                diff = dict_diff(diff['before'], diff['after'])
+        if diff:
+            diff = colorize(str(diff), 'changed')
+            print(self._indent_text(diff, indent_level+4))
+
+    def _print_host_or_item(self, host_or_item, changed, msg, diff, is_host, error, stdout, stderr):
+        if is_host:
+            indent_level = 0
+            name = colorize(host_or_item.name, 'not_so_bold')
+        else:
+            indent_level = 4
+            if isinstance(host_or_item, dict):
+                if 'key' in host_or_item.keys():
+                    host_or_item = host_or_item['key']
+            name = colorize(unicode(host_or_item), 'bold')
```

Comment:
Python 3.5 does not have `unicode`. Take a look at the [ansible text helper methods](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/_text.py) for alternatives.

Only noting the first occurrence.